src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/MediaStreamControllerAdminTest.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Tests\Functional\Controller;
+
+use Prophecy\Argument;
+use Sulu\Bundle\MediaBundle\Admin\MediaAdmin;
+use Sulu\Bundle\MediaBundle\DataFixtures\ORM\LoadCollectionTypes;
+use Sulu\Bundle\MediaBundle\DataFixtures\ORM\LoadMediaTypes;
+use Sulu\Bundle\MediaBundle\Entity\Collection;
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+use Sulu\Component\Security\Authorization\PermissionTypes;
+use Sulu\Component\Security\Authorization\SecurityCheckerInterface;
+use Sulu\Component\Security\Authorization\SecurityCondition;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class MediaStreamControllerAdminTest extends SuluTestCase
+{
+    /**
+     * @var KernelBrowser
+     */
+    private $client;
+
+    public function setUp(): void
+    {
+        $this->client = $this->createAuthenticatedClient();
+        $this->purgeDatabase();
+
+        $collectionTypes = new LoadCollectionTypes();
+        $collectionTypes->load($this->getEntityManager());
+        $mediaTypes = new LoadMediaTypes();
+        $mediaTypes->load($this->getEntityManager());
+    }
+
+    public function testDownloadActionCheckPermissionCalled()
+    {
+        $filePath = $this->createMediaFile('test.jpg');
+        $media = $this->createMedia($filePath, 'file-without-extension');
+
+        // teardown needed to set a mocked service
+        $this->tearDown();
+        $this->client = $this->createAuthenticatedClient();
+
+        $securityChecker = $this->prophesize(SecurityCheckerInterface::class);
+        $securityChecker->checkPermission(Argument::that(function(SecurityCondition $securityCondition) use ($media) {
+            $this->assertSame(MediaAdmin::SECURITY_CONTEXT, $securityCondition->getSecurityContext());
+            $this->assertSame(null, $securityCondition->getLocale());
+            $this->assertSame(Collection::class, $securityCondition->getObjectType());
+            $this->assertSame($media->getCollection(), $securityCondition->getObjectId());
+
+            return true;
+        }), PermissionTypes::VIEW)
+            ->shouldBeCalled();
+
+        self::$container->set('sulu_security.security_checker', $securityChecker->reveal());
```

Comment:
Shouldn't we make that a fully functional tests? And same for the same controller in the website context? Then the bug that downloading still works would have been recognized.

Reply:
@danrot Need to check that but our TestVoter currently does not allow to do security specific tests.

Reply:
Worst case scenario would be to do something like I already did in #5439: https://github.com/sulu/sulu/pull/5439/commits/557c5b8c227b017b050a10dd5f4e9a73b93d2af1#diff-b5ddd621eded67d076a89bcb768bef31R31

Reply:
Test added beneath